src/expr/sequence.h

```diff
@@ -131,33 +137,31 @@ class Sequence
   size_t roverlap(const Sequence& y) const;
 
   /** get the element type of the sequence */
-  TypeNode getType() const { return d_type; }
+  const TypeNode& getType() const;
   /** get the internal Node representation of this sequence */
-  const std::vector<Node>& getVec() const { return d_seq; }
+  const std::vector<Node>& getVec() const;
   /** get the internal node value of the first element in this sequence */
-  Node front() const;
+  const Node& front() const;
   /** get the internal node value of the last element in this sequence */
-  Node back() const;
+  const Node& back() const;
+  /** @return The element at the i-th position */
+  const Node& nth(size_t i) const;
   /**
    * Returns the maximum sequence length representable by this class.
    * Corresponds to the maximum size of d_seq.
    */
   static size_t maxSize();
 
-  //!!!!!!!!!!!!!!! temporary
-  ExprSequence toExprSequence();
-  //!!!!!!!!!!!!!!! end temporary
-
  private:
   /**
    * Returns a negative number if *this < y, 0 if *this and y are equal and a
    * positive number if *this > y.
    */
   int cmp(const Sequence& y) const;
   /** The element type of the sequence */
-  TypeNode d_type;
+  std::unique_ptr<TypeNode> d_type;
```

Comment:
Why the change to unique ptr here?  Is this for destruction issues?

Reply:
No, the issue is that `TypeNode` is only forward declared here (there's a circular dependency). Right now, we don't have an easy way around that. Using an `std::unique_ptr` is what we do for other such classes as well, e.g. https://github.com/CVC4/CVC4/blob/master/src/expr/array_store_all.h#L59-L60.

Reply:
Right, I figured it was something like that.  Looks good to me.